### PR TITLE
How to install on Ubuntu 16.04

### PR DESCRIPTION
**Pull request by okin**
I created instructions on how to setup Mattermost on an Ubuntu 16.04.
This was something *it33* asked for nicely ([on the forums](http://forum.mattermost.org/t/ubuntu-16-04-support/2406)).

Compared with the instructions for setting up Ubuntu 14.04 I changed the markup on some places to make things nicer to read.

### CODE REVIEW

**Comment by esethna** [comment]
Thanks @okin! This is great! Queuing for review.

File: source/install/prod-ubuntu-16.04.rst [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+2. For the sake of making this guide simple we located the files at
+   ``/home/ubuntu/mattermost``.
+3. We have also elected to run the Mattermost Server as the ``ubuntu``
+   account for simplicity. We recommend setting up and running the
+   service under a ``mattermost`` user account with limited permissions.
+4. Download `any version of the Mattermost Server <https://docs.mattermost.com/administration/upgrade.html#version-archive>`_ by typing::
+
+       wget https://releases.mattermost.com/X.X.X/mattermost-X.X.X-linux-amd64.tar.gz
+
+   Where ``X.X.X`` is typically the latest Mattermost release version, which is currently ``3.4.0``.
```
Comment (yangchen1) [inline source/install/prod-ubuntu-16.04.rst]: We are now on 3.5. :) Might want to just not mention it, so we don't have to update every time.

Comment (okin) [inline source/install/prod-ubuntu-16.04.rst]: I have updated the guide to not mention a specific version. The bullet point points to the archive overview and I think that should suffice.

File: source/install/prod-ubuntu-16.04.rst [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+
+        psql
+
+5.  Create the Mattermost database by typing::
+
+        postgres=# CREATE DATABASE mattermost;
+
+6.  Create the Mattermost user by typing::
+
+        postgres=# CREATE USER mmuser WITH PASSWORD 'mmuser_password';
```
Comment (enahum) [inline source/install/prod-ubuntu-16.04.rst]: We should add a note like the one here https://github.com/mattermost/docs/pull/573/files#diff-d3d33d958ceed9b270f0d2b400e98b77R120

File: source/install/prod-ubuntu-16.04.rst [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+2. Make sure the system is up to date with the most recent security
+   patches.
+
+   ::
+
+       sudo apt update
+       sudo apt upgrade
+
+
+Set up Database Server
```
Comment (enahum) [inline source/install/prod-ubuntu-16.04.rst]: What about MySQL?

File: source/install/prod-ubuntu-16.04.rst [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+        psql --host=10.10.10.1 --dbname=mattermost --username=mmuser --password
+        mattermost=> \q
+
+
+Set up Mattermost Server
+------------------------
+
+1. For the purposes of this guide we will assume this server has an IP
+   address of ``10.10.10.2``
+2. For the sake of making this guide simple we located the files at
```
Comment (enahum) [inline source/install/prod-ubuntu-16.04.rst]: We should be consistent with this [PR](https://github.com/mattermost/docs/pull/573) and use `mattermost` user and group

File: source/install/prod-ubuntu-16.04.rst [hunk 5]
```diff
@@ -0,0 +1,489 @@
...
+         After=postgresql.service
+         Requires=postgresql.service
+
+         [Service]
+         Type=simple
+         ExecStart=/home/mattermost/mattermost/bin/platform
+         Restart=always
+         RestartSec=10
+         WorkingDirectory=/home/mattermost/mattermost
+         User=mattermost
```
Comment (enahum) [inline source/install/prod-ubuntu-16.04.rst]: If we are using user and group `mattermost` they should be created at some point as I commented above

File: source/install/prod-ubuntu-16.04.rst (hunk 3 above)
Comment (okin) [inline source/install/prod-ubuntu-16.04.rst]: @enahum: I did not try and would not like to as I had some unpleasant experiences with changes in recent MySQL. Postgres has proven to be fairly stable in my experience and I think it is a good choice for most people following this guide.

Also the other guides for installing Mattermost all refer to postgres as database and do not explain how to setup MySQL. I'd like to stay in line with these guides.

**Comment by okin** [comment]
@enahum thanks for your feedback! I adjusted the mentioned parts to use a different user and also adapted some things from the PR you mentioned.

I did some slight rewrite in the Let's encrypt section. I previously copied it from the existing Ubuntu 14.04 guide but do not have the time to fully verify this. As far as I understand the commands things _should_ be working but it is untested and therefore a thorough check would be nice.

File: source/install/prod-ubuntu-16.04.rst (hunk 3 above)
Comment (enahum) [inline source/install/prod-ubuntu-16.04.rst]: The new installation guid for ubuntu adds the MySQL so I think this one should too, im just saying I'll leave that for the PM's to decide

https://github.com/mattermost/docs/pull/573

Comment (esethna) [inline source/install/prod-ubuntu-16.04.rst]: Thanks @enahum, I think we can take this PR as is and add MySQL in a new PR if you're interested @okin . Agree the end goal would be to have Postgres and MySQL in both install guides. PR for adding MySQL to 14.04 guide was just merged. https://github.com/mattermost/docs/pull/573

**Comment by esethna** [comment]
@okin please rebase when you have a chance?

**Comment by okin** [comment]
@esethna did a rebase :)

**Comment by esethna** [comment]
Thanks @okin, much appreciated!!

**Comment by esethna** [comment]
HI @okin, question on this,

In our other install guides we've had issues with the **ulimit configuration**, I just want to verify that this is covered or not necessary in the 16.04 install guide?

See screenshot example from Ubunu 14.04 guide:

image

**Comment by okin** [comment]
Hi Eric, 
this is currently not covered. Seems to have escaped my view! 

After a quick search I think adding the following to the service file in the `[Service]`-Section should do the trick:

 LimitNOFILE=50K 50K

(It is slightly more but I think readability is nicer.) 

Do you have any instance (my one just carries around 10 users) or an easy way to test this?

-Niko 

Am 28. November 2016 20:59:04 MEZ, schrieb Eric Sethna <notifications@github.com>:

**Pull request merged**